onnxruntime/core/providers/cpu/tensor/slice.cc

```diff
@@ -79,59 +107,160 @@ Status SliceBase::PrepareForCompute(const std::vector<int64_t>& raw_starts,
   }
 
   // Iterate through the provided axes and override the start/end ranges
+  std::unordered_set<int64_t> unique_axes;
   const auto& dimension_count = input_dimensions.size();
   for (size_t axesIndex = 0; axesIndex < axes.size(); axesIndex++) {
     auto axis = axes[axesIndex] < 0 ? axes[axesIndex] + static_cast<int64_t>(dimension_count) : axes[axesIndex];
     if (axis >= static_cast<int64_t>(dimension_count) || axis < 0)
       return Status(ONNXRUNTIME, INVALID_ARGUMENT, "'axes' has an axis outside of the tensor dimension count");
+    if (unique_axes.find(axis) != unique_axes.end())
+      return Status(ONNXRUNTIME, INVALID_ARGUMENT, "'axes' has duplicates");
+    unique_axes.insert(axis);
+
+    // process start
     auto start = raw_starts[axesIndex];
     if (start < 0)
       start += input_dimensions[axis];
     starts[axis] = clamp(start, int64_t{0}, input_dimensions[axis]);
 
+    // process end
     auto end = raw_ends[axesIndex];
     if (end < 0)
       end += input_dimensions[axis];
-    output_dims[axis] = clamp(end, int64_t{0}, input_dimensions[axis]) - starts[axis];
-    if (output_dims[axis] < 0)
-      return Status(ONNXRUNTIME, INVALID_ARGUMENT, "'starts' and 'ends' values resulted in a negative dimension");
+
+    // find output dim value for this axis
+    auto temp = clamp(end, int64_t{0}, input_dimensions[axis]) - starts[axis];
+    if (temp < 0)
+      output_dims[axis] = 0;
+    else
+      output_dims[axis] = temp;
   }
 
   return Status::OK();
 }
 
+// DynamicSlice & Slice V10
+Status SliceBase::PrepareForCompute(const std::vector<int64_t>& raw_starts,
+                                    const std::vector<int64_t>& raw_ends,
+                                    const std::vector<int64_t>& raw_axes,
+                                    const std::vector<int64_t>& raw_steps,
+                                    const std::vector<int64_t>& input_dimensions,
+                                    std::vector<int64_t>& starts,
+                                    std::vector<int64_t>& steps,
+                                    std::vector<int64_t>& output_dims) const {
+  // Initialize axes to the provided axes attribute or to the default sequence
+  std::vector<int64_t> axes(raw_axes);
+  if (axes.size() == 0) {
```

Comment:
```suggestion
  if (!axes.empty()) {
```

Reply:
changed to axes.empty() as the previous line had `axes.size() == 0`